src/component/home/index.tsx

```diff
@@ -219,11 +220,18 @@ export function Component(props: HomeProps) {
       }}
       timelineButton={{
         "aria-label": "Toggle sidebar",
-        onPress: () => handleSubmit(handleAddTimeline)(),
+        onPress: () => {
+          if (!hasValue) {
+            setShowSidebar(!showSidebar);
+          } else {
+            handleSubmit(handleAddTimeline)();
+          }
+        },
```

Comment:
Just for awareness because I see this more often and I start to realize where to draw the line. This here mixes the presentation layer with certain state management and business logic. The presentation layer part is the react component itself. The state management is the state of "has value" or "show sidebar". The business logic evolves around managing this state but could be more, e.g. if more data is involved that has to be fetched from an API. So maybe we could come up with a pattern at some point in the future to separate this a little better. Then we would actually be able to write unit tests for the state changes. Having hooked up the business logic with the react component would then be more straight forward and we would have achieved some level of automated verification of presentation layer management.